### DISCUSSION TITLE
Can it compile to a dll/so?

### DISCUSSION THREAD

**Discussion by Zireael07** [titzer/virgil#103] [Q&A]
Title.

**Comment by titzer** [comment]
Currently no. The Virgil compiler always produces a statically linked executable. Would your use case need linking a .so?

**Comment by Zireael07** [comment]
Yes (it's for a preexisting game engine that can use external dll's for scripting languages support. No hurry though, as I don't expect their docs/API to be finalized in a couple months from now

**Comment by titzer** [comment]
I don't have the bandwidth on that atm, but off the top of my head, it would require emitting an additional section for the dynamic linker and some internal compiler work to deal with addresses that aren't fixed.

**Comment by crystalthoughts** [comment]
If I'm not mistaken (I'm not yet really a systems coder), this would allow for development of numerical libraries in virgil, that could be called by other code (e.g a GUI) written in another language? If so this could help boost adoption quite a bit if it was advertised that way. I've spent a lot of time looking into ways of avoiding C++ :)

**Reply by titzer** [reply]
>  I've spent a lot of time looking into ways of avoiding C++ :)

Heh, yes. Well you have come to the right place to avoid the C world :-)

I could use some help on making linkable `.so` files. As I mentioned above, it'd require emitting another section in the ELF file, potentially making it position-independent, and adding symbols. Currently, symbols are supported, but maybe an additional name mangling step to work.
